function rsReconfigMember(member){
  var cfg = rs.config()
  cfg.members.forEach(function(part,index,memberArray){
    if (member.host == part.host) {
      for(k in member){
        memberArray[index][k] = member[k]
      }
    }
  })
  return rs.reconfig(cfg)
}

function rsReconfigSettings(settings){
  var cfg = rs.config()
  cfg.settings = settings
  return rs.reconfig(cfg)
}

function authRequired() {
  try {
    return db.serverCmdLineOpts().code == 13;
  } catch (err) {
    return false;
  }
}

if (authRequired()) {
  try {
<% if @replset -%>
    rs.slaveOk()
<% end -%>
    var prev_db = db
    db = db.getSiblingDB('admin')
    db.auth('<%= @admin_username %>', '<%= @admin_password %>')
    db = db.getSiblingDB(prev_db)
  }
  catch (err) {
    // This isn't catching authentication errors as I'd expect...
    abort('Unknown error')
  }
}
